python/lsst/sims/catUtils/mixins/PhotometryMixin.py

```diff
@@ -148,6 +148,35 @@ def get_lsst_photometric_uncertainties(self):
                                                  'lsstBandpassDict')
 
 
+    def calculateVisibility(self, magFilter, sigma=0.1, randomSeed=None):
+        """
+        Determine (probabilistically) whether a source was detected or not.
+
+        The 'completeness' of source detection at any magnitude is calculated by
+          completeness = (1 + e^^(magFilter-m5)/sigma)^(-1)
+        For each source with a magnitude magFilter, if a uniform random number [0-1)
+          is less than or equal to 'completeness', then it is counted as "detected".
+        See equation 24, figure 8 and table 5 from SDSS completeness analysis in
+        http://iopscience.iop.org/0004-637X/794/2/120/pdf/apj_794_2_120.pdf
+        "THE SLOAN DIGITAL SKY SURVEY COADD: 275 deg2 OF DEEP SLOAN DIGITAL SKY SURVEY IMAGING ON STRIPE 82"
+
+        @ param [in] magFilter is the magnitude of the object in the observed filter.
+
+        @ param [in] sigma is the FWHM of the distribution
+
+        @ param [out] visibility (None/1).
+        """
+        if len(magFilter) == 0:
+            return numpy.array([])
+        # Calculate the completeness at the magnitude of each object.
+        completeness = 1.0 / (1 + numpy.exp((magFilter - self.obs_metadata.m5[self.obs_metadata.bandpass])/sigma))
+        if randomSeed is not None:
+            numpy.random.seed(randomSeed)
```

Comment:
Note: if your query to fatboy is large enough, you could get your results in several chunks, in which case this method will be called once for each chunk, and the seed will keep getting reset.

You can have a member variable that exists just to check whether or not numpy.random has been seeded.  Something like:

if not hasattr(self, '_ssm_has_been_seeded'):
    numpy.random.seed(obs_metadata.mjd)
    self._ssm_has_been_seeded = True

Then the sequence will only be seeded once.